### PR TITLE
CloudEvent

### PR DESCRIPTION
**Pull request by ritishgumber**

### CODE REVIEW

File: src/CloudEvent.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
+import CB from './CB'
+import axios from 'axios'
+var os = require('os');/*
+ CloudEvent
+ */
+class CloudEvent {
+    constructor() {
```
Comment (nawazdhandala) [inline src/CloudEvent.js]: We dont need a constructor.

File: src/CloudEvent.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                        def.reject(err);
+                    }
+                }
+            });
+        }
+    })
+
+}
+
+CloudEvent.getDeviceInformation = function(callback) {
```
Comment (nawazdhandala) [inline src/CloudEvent.js]: This should be a private function.

File: src/CloudEvent.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        obj['browser'] = 'node';
+
+    _getLocation(obj, {
+        success: function(object) {
+            callback.success(object)
+        }
+    })
+}
+
+CloudEvent.os = function() {
```
Comment (nawazdhandala) [inline src/CloudEvent.js]: Private function.

File: src/CloudEvent.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+    }
+    if (!type)
+        type = 'Custom';
+    if (!callback)
+        def = new CB.Promise();
+
+    CloudEvent.getDeviceInformation({
+        success: function(object) {
+            data['deviceInfo'] = object;
```
Comment (nawazdhandala) [inline src/CloudEvent.js]: Just say device. :)

File: src/CloudEvent.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+var os = require('os');/*
+ CloudEvent
+ */
+class CloudEvent {
+    constructor() {
+        this.name = '';
+        this.data = {};
+        this.type = 'Custom';
+        this.id = null;
+        this.ACL = new CB.ACL(); //ACL(s) of the document
```
Comment (shubhamqweasd) [inline src/CloudEvent.js]: why we need ACL here , when we already save a cloudObject in track method. ?

File: src/CloudEvent.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                    }
+                },
+                error: function(err) {
+                    if (callback) {
+                        callback.error(err);
+                    } else {
+                        def.reject(err);
+                    }
+                }
+            });
```
Comment (shubhamqweasd) [inline src/CloudEvent.js]: we dont need to return def Promise here ?

**Pull request closed without merging**